src/App/Controller/PrivateDiscussionController.php

```diff
@@ -0,0 +1,325 @@
+                    'slug' => $discussion->getSlug(),
+                ]));
+            }
+
+            return $this->render('App:PrivateDiscussion:show.html.twig', [
+                'discussion' => $discussion,
+                'form'       => $form->createView(),
+                ]);
+        }
+
+        return $this->render('App:PrivateDiscussion:show.html.twig', [
+            'discussion' => $discussion,
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function addAction(Request $request)
+    {
+        $discussion = new PrivateDiscussion();
+        $user = $this->getUser();
+        $form = $this->createForm(new AddPrivateDiscussionType($user->getId()), $discussion);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $discussion->setAdmin($user)->resetUnreaders();
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('private_discussion_show', [
+                'slug' => $discussion->getSlug(),
+            ]));
+        }
+
+        return $this->render('App:PrivateDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function editAction(Request $request, $slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PrivateDiscussion')->findOneBySlug($slug);
+
+        if (null ===  $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        if (false === $this->get('security.context')->isGranted('edit', $discussion)) {
+            throw new AccessDeniedException();
+        }
+
+        $form = $this->createForm(new EditDiscussionType(), $discussion);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($discussion);
+            $em->flush();
+
+            return $this->redirect($this->generateUrl('private_discussion_show', [
+                'slug' => $discussion->getSlug(),
+             ]));
+        }
+
+        return $this->render('App:PrivateDiscussion:form.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Security("has_role('ROLE_USER')")
+     */
+    public function addMemberAction(Request $request, $slug)
+    {
+        $discussion = $this->getDoctrine()->getRepository('App:PrivateDiscussion')->findOneBySlug($slug);
+
+        if (null ===  $discussion) {
+            throw $this->createNotFoundException();
+        }
+
+        $users         = $this->getDoctrine()->getRepository('App:User')->findAll();
+        $usersNumber   = count($users);
+        $members       = $discussion->getMembers();
+        $membersNumber = count($members);
+
+        if ($usersNumber === $membersNumber) {
```

Comment:
If you don't need these variables elsewhere, you can do directly `if (count($users) === count($members))`.